src/components/lib/protdom/struct_defs_arch/x86_64/protdom_stuct_def.h

```diff
@@ -0,0 +1,37 @@
+
+#include <cos_types.h>
+#include <cos_kernel_api.h>
+
+#define PROTDOM_VAS_NAME_SZ 	(1ULL << 39)
+#define PROTDOM_VAS_NUM_NAMES 	256
+#define PROTDOM_MPK_NUM_NAMES 	14
+#define PROTDOM_ASID_NUM_NAMES 	4096
+
+#define PROTDOM_NS_STATE_RESERVED 	1
+#define PROTDOM_NS_STATE_ALLOCATED 1 << 1
+#define PROTDOM_NS_STATE_ALIASED 	1 << 2
+
+
+struct protdom_vas_name {
+	u32_t state : 3;
```

Comment:
Generally, we want to avoid bitfields, but they can be OK. Here, since you don't have multiple fields that are bitfields, this is really no different than simply having both a u32_t and a pointer in the struct. Don't think they're helping much here, and they are making the code slightly more complex. I might be missing something about their usage.